### PR TITLE
Don't forget validator IDs immediately when a message fails to send

### PR DESCRIPTION
**Pull request by hrxi**
Touches upon #2294 but does not address the root cause.

### CODE REVIEW

File: validator-network/src/network_impl.rs [hunk 1]
```diff
@@ -90,102 +113,105 @@ where
...
+            .get_mut(public_key.compressed())
+        {
+            Some(cache_entry) => {
+                if let CacheState::InProgress(prev_peer_id) = *cache_entry {
+                    *cache_entry = match cache_value {
+                        Ok(peer_id) => CacheState::Resolved(peer_id),
+                        Err(()) => CacheState::Error(prev_peer_id),
+                    };
+                } else {
+                    unreachable!("cache state must be \"in progress\"");
```
Comment (jsdanielh) [inline validator-network/src/network_impl.rs:156]: I'd add a comment that this function assumes that the cache entry is updated with an `InProgress` entry before.

File: validator-network/src/network_impl.rs [hunk 2]
```diff
@@ -90,102 +113,105 @@ where
...
-                        .insert(public_key.compressed().clone(), CacheState::InProgress);
-                    log::debug!(
-                        ?public_key,
-                        validator_id,
-                        "Could not find peer ID for validator in DHT. Performing a DHT query",
-                    );
-                }
+                    CacheState::Empty(prev_peer_id) => CacheState::InProgress(Some(prev_peer_id)),
+                };
+                *cache_state = new_cache_state.clone();
```
Comment (jsdanielh) [inline validator-network/src/network_impl.rs]: Clippy warning:
```suggestion
                *cache_state = new_cache_state;
```

File: validator-network/src/network_impl.rs [hunk 3]
```diff
@@ -90,102 +113,105 @@ where
...
+            } else {
+                new_cache_state = CacheState::InProgress(None);
+                // No cache entry for this validator ID: we are going to perform the DHT query
+                validator_peer_id_cache
+                    .insert(public_key.compressed().clone(), new_cache_state.clone());
```
Comment (jsdanielh) [inline validator-network/src/network_impl.rs]: Clippy warning:
```suggestion
                    .insert(public_key.compressed().clone(), new_cache_state);
```

File: validator-network/src/network_impl.rs (hunk 1 above)
Comment (hrxi) [reply validator-network/src/network_impl.rs:156]: Done.

File: validator-network/src/network_impl.rs (hunk 2 above)
Comment (hrxi) [reply validator-network/src/network_impl.rs]: Fixed.

File: validator-network/src/network_impl.rs (hunk 3 above)
Comment (hrxi) [reply validator-network/src/network_impl.rs]: Fixed.

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**